taskcluster/docker/node/build.py

```diff
@@ -150,20 +189,35 @@ def check_manifest(path, buildid_version):
                 continue
             _found = True
             if not _id.endswith(ID_ALLOWLIST):
-                raise Exception(f"{_key}.gecko.id {_id} must end with one of the following suffixes!\n{ID_ALLOWLIST}")
+                raise Exception(
+                    f"{_key}.gecko.id {_id} must end with one of the following suffixes!\n{ID_ALLOWLIST}"
+                )
             else:
                 print(f"Add-on id {_id} matches the allowlist.")
         if manifest["version"] != buildid_version:
-            raise Exception(f"{manifest['version']} doesn't match buildid version {buildid_version}!")
+            raise Exception(
+                f"{manifest['version']} doesn't match buildid version {buildid_version}!"
+            )
+        if not is_version_mv3_compliant(manifest["version"]):
```

Comment:
If the intent is to restrict this check to MV3 extensions, we should check the value of the `manifest_version` prop first. Otherwise, this function should be renamed to `is_simplified_version`.